### PR TITLE
Convert class to subtype and introduce class in buildings

### PR DESCRIPTION
**Pull request by jenningsanderson**
# Description

The _building_ type is the only Overture type that includes `class` without a `subtype`. This is mostly because the building schema was decided before other themes started using the term `subtype`. 

A backlog issue for buildings has been to promote `class` to `subtype` to match the rest of Overture. 

Additionally, the current allowed `class` values are not granular enough for specific rendering cases, primarily parking garages. There is only `class=transportation`, which includes more than just parking.

This PR promotes `class` to `subtype` and introduces new `class` values for buildings. The majority of the time, the `class` tag should simply be the value of OSM's `building` tag.

This allows anyone to use the current `class` types as `subtype` (backwards compatible) and introduces new `class` values that allow for more detail, like `parking`.

# Reference

*List of relevant links to GitHub issues, PRs, and other documentation.*

1. TODO.

# Testing

*Brief description of the testing done for this change showing why you are confident it works as expected and does not introduce regressions. Provide sample output data where appropriate.* 

TODO.

# Checklist

*Checklist of tasks commonly-associated with schema pull requests. Please review the relevant checklists and ensure you do all the tasks that are required for the change you made.*

1. [x] Add relevant examples.
4. [ ] Add relevant counterexamples.
5. [x] Update any counterexamples that became obsolete. For example, if a counterexample uses property `A` but is not intended to test property `A`'s validity, and you made a schema change that invalidates property `A` in that counterexample, fix the counterexample to align it with your schema change.  
6. [x] Update in-schema documentation using plain English written in complete sentences, if an update is required.
7. [ ] Update Docusaurus documentation, if an update is required.
8. [ ] Review change with Overture technical writer to ensure any advanced documentation needs will be taken care of, unless the change is trivial and would not affect the documentation.

# Documentation Website

[Docs preview for this PR.](https://dfhx9f55j8eg5.cloudfront.net/pr/183)

### CODE REVIEW

**Review by jwass — DISMISSED** [review]
This sounds good to me.

File: schema/buildings/defs.yaml [hunk 1]
```diff
@@ -90,7 +90,7 @@ shapeContainer:
       description: >-
         Bearing of the roof ridge line in degrees.
       type: number
-      exclusiveMinimum: 0
+      inclusiveMinimum: 0
```
Comment (jenningsanderson) [inline schema/buildings/defs.yaml:93]: 0 is an allowable direction for a roof (North)

**Review by vcschapp — DISMISSED** [review]
Excellent to see the improve alignment with other themes.

**Review by jwass — DISMISSED** [review]

**Review by DavidKarlas — APPROVED** [review]

File: schema/buildings/building.yaml [hunk 2]
```diff
@@ -25,27 +25,123 @@ properties:
...
+          - sty
+          - supermarket
+          - synagogue
+          - temple
+          - terrace
+          - toilets
+          - townhouse
+          - train_station
+          - transformer_tower
+          - transportatio
```
Comment (vcschapp) [inline schema/buildings/building.yaml]: oh ohhhhhh!

File: schema/buildings/building.yaml [hunk 3]
```diff
@@ -25,27 +25,123 @@ properties:
...
+          - house
+          - houseboat
+          - hut
+          - industrial
+          - kindergarten
+          - kiosk
+          - manufacture
+          - marketplace
           - military
+          - monastary
```
Comment (vcschapp) [inline schema/buildings/building.yaml]: Dana caught this.

File: schema/buildings/building.yaml [hunk 4]
```diff
@@ -25,27 +25,123 @@ properties:
...
+          - mosque
+          - office
+          - outbuilding
+          - parking
+          - pavillion
```
Comment (vcschapp) [inline schema/buildings/building.yaml]: Dana.......

File: schema/buildings/building.yaml [hunk 5]
```diff
@@ -25,27 +25,123 @@ properties:
...
+          - school
+          - semi
+          - semidetached_house
+          - service
+          - shed
+          - shop
+          - shrine
+          - silo
+          - slurry_tank
+          - sports_centre
```
Comment (vcschapp) [inline schema/buildings/building.yaml:123]: Are we doing American English?

File: schema/buildings/building.yaml [hunk 6]
```diff
@@ -25,27 +25,123 @@ properties:
...
+          - residential
+          - service
           - transportation
-          - medical
-          - entertainment
+      class:
+        description: >-
+          Further delineation of the building's built purpose.
+        type: string
+        enum:
```
Comment (LukaszMichalski-TomTom) [inline schema/buildings/building.yaml:54]: Does this list also include values from `amenity`, `turism` or other tags that are used in mapping building to its `subtype`?

Comment (jenningsanderson) [reply schema/buildings/building.yaml:54]: No, it _only_ takes values from `building` tag.

**Review by vcschapp — APPROVED** [review]

**Review by TristanDiet-TomTom — APPROVED** [review]

**Review by DavidKarlas — APPROVED** [review]

**Pull request merged**